### ISSUE TITLE
Move all tmp dirs from repos like ck-caffe/program/caffe-classification/tmp to ~/CK/local/tmp

### ISSUE THREAD

**Issue by DVEfremov** [dividiti/ck-caffe#29]
I've found some tmp dirs at reposotories 
~/CK/ck-caffe/program/caffe-classification/tmp 
but also there is some local storage with some local specific information 
~/CK/local/env

it better to keep all local specific temporary  information at one place for example 
~/CK/local/env (as it is now)
and 
~/CK/local/tmp (should be implemented)

**Comment by gfursin** [comment]
Yes, this is because it is a CK 'program' entry. For such entries, CK creates a tmp dir inside this entry to keep temporal object files, temporal deps, produced binaries, etc. This was done on purpose, so that when a user compiles and runs a given program, it's easier to look at produced files in the same directory. Also you can remove all such tmp in one go:
$ ck clean_tmp program
But eventually, if the community will find it more convenient to use another entry, why not. I just would not bother with that right now ...

**Comment by dsavenko** [comment]
Actually, CK/local/env is not very convenient right now, as the subdirs are named after UIDs, and it's hard to find the one you need without resolving UIDs via ck or grep'ing. So, I'd like the 'tmp' dirs approach more. At least, until something is done to local/env to make it 'human-browseable'.

**Comment by gfursin** [comment]
Wait - CK/local/env is the registration of software versions and has nothing to do with 'tmp'! We use UID there on purpose because there is no other way - it's otherwise extremely difficult to find a nice schema for naming which would keep all dependencies with versions and host/targets (we already have this issue with CK_TOOLS and I even added an option to actually build tools inside env). Besides when we have paths more than 256 characters, we start having problem on Windows! That's why we use in CK/local/env with UID and search by meta. And there is a user friendly way to view them:
$ ck show env
You can even prune by --tags or by --target_os ...

But again, it has nothing to do with CK/local/tmp for programs ...
And I agree, we can use similar approach but likely using new module like 'program.tmp' - in such case we will not be mixing up global tmp entries and specific tmp entries for program. In such case we can even reuse program alias/UID in 'program.tmp'. But again, it's not the priority right now ...
